### PR TITLE
Update timestamp in Phase Score header

### PR DESCRIPTION
**Pull request by nickgraz**
- Update Score Phase header to use ISO date string

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2366** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2366?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karthikbtf** [comment]
@nickgraz should we be using existing date formats throughout the app? Maybe worth asking design.
We currently use '-' instead of '/' as the separator.

**Review by karthikbtf — APPROVED** [review]
Looks good but maybe we should use the same date format as in other parts of the app.
https://github.com/terraware/web-components/blob/main/src/utils/index.ts#L7

**Comment by nickgraz** [comment]
@karthikbtf I asked [in Figma](https://www.figma.com/file/Ec5QnRsnzqEKeifzwPKR3k?node-id=1667:34404&mode=design#742770074) about this, I think for now we will use the numerical representation for the locale, which seems correct. I will ask about the rest of the app, I agree, if we are using ISO as standard everywhere else we should probably use that here also to avoid confusion.

**Comment by karthikbtf** [comment]
> @karthikbtf I asked [in Figma](https://www.figma.com/file/Ec5QnRsnzqEKeifzwPKR3k?node-id=1667:34404&mode=design#742770074) about this, I think for now we will use the numerical representation for the locale, which seems correct. I will ask about the rest of the app, I agree, if we are using ISO as standard everywhere else we should probably use that here also to avoid confusion.

Sounds good @nickgraz 👍 let's go with this for now.

File: src/scenes/AcceleratorRouter/Scoring/PhaseScores/LastUpdated.tsx [hunk 1]
```diff
@@ -2,18 +2,15 @@ import React, { useMemo } from 'react';
 
 import { Typography } from '@mui/material';
 
-import { useLocalization } from 'src/providers';
 import strings from 'src/strings';
 import { PhaseScores, Score } from 'src/types/Score';
-import { getLongDate } from 'src/utils/dateFormatter';
+import { getISODate } from 'src/utils/dateFormatter';
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/Scoring/PhaseScores/LastUpdated.tsx:7]: Use `import { getDateDisplayValue } from '@terraware/web-components/utils';`?
Sorta related, plan is to move away from moment (sunset) and into luxon. web-components has both implementations just need to make the switch at some point.

**Review by karthikbtf — COMMENTED** [review]
Sorry about the back and forth on formats.

**Review by karthikbtf — APPROVED** [review]
🙏🏻

**Pull request merged**